src/main/java/RaiseTech/ControllerGet/NameController.java

```diff
@@ -0,0 +1,41 @@
+package RaiseTech.ControllerGet;
+
+
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.util.UriComponentsBuilder;
+import java.net.URI;
+import java.util.List;
+import java.util.Map;
+
+@RestController
+public  class NameController {
+    @GetMapping("/names")
+    public List<String> getNames() {
+        return List.of("koyama", "tanaka");
+    }
+
+
+    @PostMapping("/names")
+    public ResponseEntity<String> create(@RequestBody CreateForm form) {
+        // 登録処理は省略
+
+        URI url = UriComponentsBuilder.fromUriString("http://localhost:8080")
+                .path("/names/id") // id部分は実際に登録された際に発⾏したidを設定する
+                .build()
+                .toUri();
+        return ResponseEntity.created(url).body("name successfully created");
+    }
+
+    @PatchMapping("/names")
+    public ResponseEntity<Map<String, String>> update(@RequestBody CreateForm form) {
```

Comment:
CreateFormは名前から登録用のフォームということがわかると思います。
更新処理を受け付けるパスでCreateFormをつかうのは不適切ですね。
```suggestion
    public ResponseEntity<Map<String, String>> update(@RequestBody UpdateForm form) {
```

または下記のようにして登録と更新のFormを共通で利用してもいいとおもいます。
```suggestion
    public ResponseEntity<Map<String, String>> update(@RequestBody NameForm form) {
```